cibuildwheel/util.py

```diff
@@ -655,3 +672,31 @@ def chdir(new_path: Path | str) -> Generator[None, None, None]:
         yield
     finally:
         os.chdir(cwd)
+
+
+def fix_ansi_codes_for_github_actions(text: str) -> str:
+    """
+    Github Actions forgets the current ANSI style on every new line. This
+    function repeats the current ANSI style on every new line.
```

Comment:
This is weird - though I see how it could happen, GH does a lot of optimization for logs. Is there a GH ticket or commend anywhere? I've not generally noticed GH breaking elsewhere with color, but maybe most things like compilers and pytest do per-line coloring too.

Reply:
There is not. I should raise an issue somewhere. Not sure where though. Maybe I'll just raise it at actions/runner and see what they say.

Reply:
Raised an issue here: https://github.com/orgs/community/discussions/40864